experimental/terraform/README.md

```diff
@@ -1,42 +1,187 @@
+            * Choose the appropriate option for your CICD system, or choose "other" to skip CICD integration
+        * _(see `/README.md#extending-to-other-event-sources` to integrate event sources not available during setup)_
+    *   Would you like to generate mock data? (y/N)
+        *   If you select yes, a script will run through and send mock GitLab or GitHub events to your event-handler.  This will populate your dashboard with mock data.  The mock data will include the work "mock" in the source. You can generate mock data without using the setup script. See [Generating mock data](../readme.md).
+
+### Making changes
+At some point after running the setup script, you may want to make modifications to your infrastructure. Or, the Four Keys project itself may be updated with a new configuration. Terraform can be used to apply any incremental changes: after updating the configuration files, run `terraform apply`. You'll be prompted to confirm the planned changes; review them carefully, then type `yes` to proceed.
+
+### New Google Cloud projects
+
+If you've chosen to create a new Google Cloud project, after the script finishes you will have a new project with an ID in the form of `fourkeys-XXXX`. If you ever want to remove the newly created projects and all associated data, you can run `cleanup.sh`.  **Only do this when you are done experimenting with Four Keys entirely, or want to start over. Running `cleanup.sh` will remove the projects and all the collected data.**
+
+> If you want to bulk delete many projects that you've created via the setup 
+> script, all of which will be named `fourkeys-XXXX`, pass a flag to the cleanup 
+> script: `./cleanup.sh -b`
+
+### The setup explained
+The setup script does many things to help create the service architecture described in the `README.md`. These include a little bash scripting and a lot of [Terraform](https://www.terraform.io/intro/).
+
+Step by step, here's what's happening:
+1. `setup.sh` starts by collecting information from the system and the user to determine a number of configuration variables that will be provided to Terraform.
+1. If you choose to make a new project, `setup.sh` will use `gcloud` to create that project, with an ID in the form `fourkeys-XXXX`.
+1. It sets several environment variables, and writes a `terraform.tfvars` file to disk, containing inputs to Terraform.
+1. Then it invokes `install.sh`, which is responsible for provisioning the infrastructure.
+1. `install.sh` runs `gcloud builds submit` commands to build the application containers that will be used in Cloud Run services.
+1. Then it invokes Terraform, which processes the configuration files (ending in `.tf`) to provision all of the necessary infrastructure into the speficied Cloud project.
+1. If you've chosen to generate mock data, the script then calls the "data generator" python application to submit several synthetic webhook events to the event-handler service that was just created.
```

Comment:
Maybe link to the generate_data.py file?

Reply:
linked!